common/test/acceptance/tests/studio/test_studio_html_editor.py

```diff
@@ -100,3 +113,186 @@ def test_link_plugin_sets_url_correctly(self):
+        self._content_test(content)
+
+    def test_tinymce_and_codemirror_preserve_span_tags(self):
+        """
+        Scenario: TinyMCE and CodeMirror preserve span tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And type "<span>Test</span>" in the code editor and press OK
+            And I save the page
+                Then the page text contains:
+                ""
+                    <span>Test</span>
+                ""
+        """
+        content = "<span>Test</span>"
+        self._content_test(content)
+
+    def test_tinymce_and_codemirror_preserve_math_tags(self):
+        """
+        Scenario: TinyMCE and CodeMirror preserve math tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And type "<math><msup><mi>x</mi><mn>2</mn></msup></math>" in the code editor and press OK
+            And I save the page
+                Then the page text contains:
+                ""
+                    <math><msup><mi>x</mi><mn>2</mn></msup></math>
+                ""
+        """
+        content = "<math><msup><mi>x</mi><mn>2</mn></msup></math>"
+        self._content_test(content)
+
+    def test_code_format_toolbar_wraps_text_with_code_tags(self):
+        """
+        Scenario: Code format toolbar button wraps text with code tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And I set the text to "display as code" and I select the text
+            And I save the page
+                Then the page text contains:
+                ""
+                    <p><code>display as code</code></p>
+                ""
+        """
+        self.html_editor.set_text_and_select("display as code")
+        self.html_editor.click_code_toolbar_button()
+        self.html_editor.save_content()
+        html = self.container_page.content_html
+        self.assertIn('<p><code>display as code</code></p>', html)
+
+    def test_raw_hmlcomponent_does_not_change_text(self):
+        """
+        Scenario: Raw HTML component does not change text
+        Given I have created a raw HTML component
+            When I edit the page
+            And type "<li>zzzz<ol> " into the Raw Editor
+            And I save the page
+                Then the page text contains:
+                  ""
+                  <li>zzzz<ol>
+                  ""
+            And I edit the page
+                Then the Raw Editor contains exactly:
+                  ""
+                  <li>zzzz<ol>
+                  ""
+        """
+        content = "<li>zzzz<ol>"
+        self.html_editor.set_content_in_tinymce_editor(content)
+        self.html_editor.save_content()
+        html = self.container_page.content_html
+        self.assertIn(content, html)
+        self.container_page.edit()
+        editor_value = self.html_editor.editor_value
+        self.assertEqual(content, editor_value)
```

Comment:
I think we need to improve comments here so anyone can understand what the test is doing